core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1543,10 +1535,8 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
     preferredReplicaImbalanceCount.set(pric)
   }
 
-  private def triggerControllerMove(): Unit = {
-    activeControllerId.set(-1)
```

Comment:
I tried adding `onControllerResignation` in that method first, but that causes another exception to be thrown. It may be that this test needs be rethought.